### PR TITLE
Allow no handlers middleware configured properly

### PR DESCRIPTION
**Pull request by bartoszpietrzak1994**

### CODE REVIEW

File: spec/EventListener/OrderPaymentPaidListenerSpec.php [hunk 1]
```diff
@@ -19,9 +19,8 @@ function let(MessageBusInterface $commandBus): void
 
     function it_dispatches_send_invoice_email_command(MessageBusInterface $commandBus): void
     {
-        $commandBus->dispatch(Argument::that(function (SendInvoiceEmail $command): bool {
-            return $command->orderNumber() === '00000001';
-        }))->shouldBeCalled();
+        $command = new SendInvoiceEmail('00000001');
+        $commandBus->dispatch($command)->shouldBeCalled()->willReturn(new Envelope($command));
```
Comment (Zales0123) [inline spec/EventListener/OrderPaymentPaidListenerSpec.php]: `->willReturn` is sufficient, `shouldBeCalled()` should be removed

File: spec/EventProducer/OrderPaymentPaidProducerSpec.php [hunk 2]
```diff
@@ -26,12 +27,14 @@ function it_dispatches_order_payment_paid_event_for_payment(
         OrderInterface $order
     ): void {
         $payment->getOrder()->willReturn($order);
+        $order->getNumber()->willReturn('0000001');
 
-        $order->getNumber()->shouldBeCalled();
+        $dateTime = new \DateTime('now');
```
Comment (Zales0123) [inline spec/EventProducer/OrderPaymentPaidProducerSpec.php]: `'now'` is default :)

File: spec/EventListener/OrderPaymentPaidListenerSpec.php [hunk 3]
```diff
@@ -19,9 +19,8 @@ function let(MessageBusInterface $commandBus): void
 
     function it_dispatches_send_invoice_email_command(MessageBusInterface $commandBus): void
     {
-        $commandBus->dispatch(Argument::that(function (SendInvoiceEmail $command): bool {
-            return $command->orderNumber() === '00000001';
-        }))->shouldBeCalled();
+        $command = new SendInvoiceEmail('00000001');
+        $commandBus->dispatch($command)->willReturn(new Envelope($command));
```
Comment (pamil) [inline spec/EventListener/OrderPaymentPaidListenerSpec.php]: We still need to assert that, cause we do not use the return value in any way.

```suggestion
        $commandBus->dispatch($command)->shouldBeCalled()->willReturn(new Envelope($command));
```

File: spec/EventProducer/OrderPaymentPaidProducerSpec.php [hunk 4]
```diff
@@ -26,12 +27,14 @@ function it_dispatches_order_payment_paid_event_for_payment(
...
 
-        $order->getNumber()->shouldBeCalled();
+        $dateTime = new \DateTime();
+        $dateTimeProvider->__invoke()->willReturn($dateTime);
 
-        $dateTimeProvider->__invoke()->willReturn(new \DateTime('now'));
+        $event = new OrderPaymentPaid('0000001', $dateTime);
 
-        $eventBus->dispatch(Argument::type(OrderPaymentPaid::class))->shouldBeCalled();
+        $eventBus->dispatch($event)->willReturn(new Envelope($event));
```
Comment (pamil) [inline spec/EventProducer/OrderPaymentPaidProducerSpec.php]:
```suggestion
        $eventBus->dispatch($event)->shouldBeCalled()->willReturn(new Envelope($event));
```

File: spec/EventProducer/OrderPlacedProducerSpec.php [hunk 5]
```diff
@@ -24,47 +26,35 @@ function let(MessageBusInterface $eventBus, DateTimeProvider $dateTimeProvider):
...
-                    $event->date()->format(\DateTimeInterface::RFC3339_EXTENDED) === $dateTime->getWrappedObject()->format(\DateTimeInterface::RFC3339_EXTENDED)
-                ;
-            }))
-            ->shouldBeCalled()
-        ;
+        $postPersistEvent = new LifecycleEventArgs($order->getWrappedObject(), $objectManager->getWrappedObject());
+        $orderPlacedEvent = new OrderPlaced('000666', $dateTime);
 
-        $this->postPersist($event);
+        $eventBus->dispatch($orderPlacedEvent)->willReturn(new Envelope($orderPlacedEvent));
```
Comment (pamil) [inline spec/EventProducer/OrderPlacedProducerSpec.php]:
```suggestion
        $eventBus->dispatch($orderPlacedEvent)->shouldBeCalled()->willReturn(new Envelope($orderPlacedEvent));
```

File: spec/EventProducer/OrderPlacedProducerSpec.php [hunk 6]
```diff
@@ -75,17 +65,12 @@ function it_dispatches_an_order_placed_event_for_updated_order(
...
-                    $event->date()->format(\DateTimeInterface::RFC3339_EXTENDED) === $dateTime->getWrappedObject()->format(\DateTimeInterface::RFC3339_EXTENDED)
-                ;
-            }))
-            ->shouldBeCalled()
-        ;
+        $postUpdateEvent = new LifecycleEventArgs($order->getWrappedObject(), $entityManager->getWrappedObject());
+        $orderPlacedEvent = new OrderPlaced('000666', $dateTime);
 
-        $this->postUpdate($event);
+        $eventBus->dispatch($orderPlacedEvent)->willReturn(new Envelope($orderPlacedEvent));
```
Comment (pamil) [inline spec/EventProducer/OrderPlacedProducerSpec.php]:
```suggestion
        $eventBus->dispatch($orderPlacedEvent)->shouldBeCalled()->willReturn(new Envelope($orderPlacedEvent));
```

**Pull request merged**